model/task/task.go

```diff
@@ -158,8 +158,23 @@ type Task struct {
 	ExecutionPlatform ExecutionPlatform `bson:"execution_platform,omitempty" json:"execution_platform,omitempty"`
 
 	// The version of the agent this task was run on.
-	AgentVersion string                 `bson:"agent_version,omitempty" json:"agent_version,omitempty"`
-	TaskOutput   *taskoutput.TaskOutput `bson:"task_output,omitempty" json:"task_output,omitempty"`
+	AgentVersion string `bson:"agent_version,omitempty" json:"agent_version,omitempty"`
+	// TaskOutputInfo holds the information for the interface that
+	// coordinates persistent storage of a task's output data.
+	// There are four possible scenarios:
+	//     1. The task will never have output data (e.g., display tasks)
+	//        and, therefore, the value is and always will be nil.
+	//     2. The task does not have output data yet, but can in the future
+	//        after running, and, therefore, the value is currently nil.
+	//     3. The task ran with the task output information initialized
+	//        upon getting dispatched and the application can safely use
+	//        this field to fetch any output data.
+	//     4. The task has data but was run before the introduction of the
+	//        field and should be initialized to the zero value on the
+	//        application to safely fetch any output data.
```

Comment:
I would rephrase this since zero value is the nil pointer in this case, which does not require initialization to set. This is already the value for all tasks run before this field existed.